src/main/java/com/pkb/unit/tracker/Tracker.java

```diff
@@ -26,6 +32,51 @@
 public class Tracker {
     private Tracker() {}
 
+    private static class RestartTracker implements ObservableSource<Boolean> {
+        private boolean hasStopped;
+        private boolean hasStarted;
+
+        private Disposable disposable;
+
+        private Observer<? super Boolean> observer;
+
+        RestartTracker(Bus bus, String id) {
+            hasStopped = false;
+            hasStarted = false;
+            disposable = payloads(bus.events(), Transition.class, id)
+                    .subscribe(this::onTransition);
+        }
+
+        private void onTransition(Transition x) {
+            if (!hasStopped && x.current() == STOPPED) {
+                hasStopped = true;
+            } else if (hasStopped && x.current() == STARTED) {
+                hasStarted = true;
+                disposable.dispose();
+                observer.onNext(true);
+                observer.onComplete();
+            }
+        }
+
+        @Override
+        public void subscribe(Observer<? super Boolean> observer) {
+            this.observer = observer;
+            observer.onSubscribe(disposable);
+        }
+    }
+
+    /**
+     * Creates an observable that emits 'true' when a given unit has transitioned from
+     * started -> stopped -> started. This can be useful to ensure that a restart
+     * has taken place when desired, and that the unit has returned to an available state.
+     * @param bus the bus to observe for events
+     * @param id the ID of the unit to be observed for a restart
+     * @return An observable, which emits true & completes when the desired unit has restarted.
+     */
+    public static Observable<Boolean> unitRestarted(Bus bus, String id) {
```

Comment:
Am I right that this `Observable` won't emit true if the client calls `unitRestarted()` after the started -> stopped transition? If yes then it might be a good idea to put it into the docs that 

> The returned observable won't emit true for previous or ongoing restarts.

Reply:
yes that's true. Good point.